core/vm/interpreter.go

```diff
@@ -138,6 +134,11 @@ func (in *EVMInterpreter) Run(contract *Contract, input []byte) (ret []byte, err
 	in.evm.depth++
 	defer func() { in.evm.depth-- }()
 
+	if readOnly && !in.readOnly {
```

Comment:
Please add comment that was in the previous code:
```
	// Make sure the readonly is only set if we aren't in readonly yet
	// this makes also sure that the readonly flag isn't removed for
	// child calls.
```

Reply:
The comment has been added back.